From ba6ce9ce5ca7243af137dbda72fee76ee46094f6 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Thu, 29 May 2014 10:01:03 +0300
Subject: [PATCH 177/213] Version 0.9.7.1: Solves dam SQLiteException in
 Datahandler

Solves the SQLiteException when fetching dam data from database
---
 MistroFarmer/src/main/AndroidManifest.xml                             | 4 ++--
 .../main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java   | 2 +-
 .../org/cgiar/ilri/mistro/farmer/backend/database/DatabaseHelper.java | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/MistroFarmer/src/main/AndroidManifest.xml b/MistroFarmer/src/main/AndroidManifest.xml
index 375fae8..a29a593 100644
--- a/MistroFarmer/src/main/AndroidManifest.xml
+++ b/MistroFarmer/src/main/AndroidManifest.xml
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="org.cgiar.ilri.mistro.farmer"
-    android:versionCode="7"
-    android:versionName="0.9.7" >
+    android:versionCode="8"
+    android:versionName="0.9.7.1" >
 
     <uses-sdk
         android:minSdkVersion="7"
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
index 1df1e3f..15fc5bf 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/DataHandler.java
@@ -684,7 +684,7 @@ public class DataHandler
                     //set dam
                     if(cowResult[cowIndex][8].length() > 0){
                         columns = new String[]{"id", "name", "ear_tag_number", "date_of_birth", "age", "age_type", "sex", "sire_id", "dam_id", "date_added", "service_type", "country_id", "bull_owner", "owner_name"};
-                        selection = "id="+cowResult[cowIndex][7];
+                        selection = "id="+cowResult[cowIndex][8];
                         String[][] damRes = databaseHelper.runSelectQuery(readableDB, databaseHelper.TABLE_COW, columns, selection, null, null, null, null, null);
                         if(damRes.length == 1){
                             Dam dam = new Dam();
diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/database/DatabaseHelper.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/database/DatabaseHelper.java
index 690cc05..1af4502 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/database/DatabaseHelper.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/backend/database/DatabaseHelper.java
@@ -18,7 +18,7 @@ import java.io.Serializable;
 public class DatabaseHelper extends SQLiteOpenHelper implements Serializable{
 
     public static final String DB_NAME = "ngombe_planner";
-    public static final int DB_VERSION = 10;
+    public static final int DB_VERSION = 11;
     public static final String TABLE_FARMER = "farmer";
     public static final String TABLE_COW = "cow";
     public static final String TABLE_EVENT = "event";
-- 
2.7.1

